### PR TITLE
Cherry-pick: dialog shortcut fixes to 14.6

### PR DESCRIPTION
**Pull request by pleku**

### CODE REVIEW

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
 import com.vaadin.flow.component.HasComponents;
 import com.vaadin.flow.component.HasSize;
 import com.vaadin.flow.component.HasStyle;
+import com.vaadin.flow.component.HasTheme;
```
Comment (sosa-vaadin) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:33]: This file imports yet does not use the `HasTheme` interface. Please remove from imports.

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 2]
```diff
@@ -623,9 +626,47 @@ public Registration addDetachListener(
...
+     * @param variants
+     *            theme variants to remove
+     */
+    public void removeThemeVariants(DialogVariant... variants) {
+        getThemeNames()
+                .removeAll(Stream.of(variants)
+                        .map(DialogVariant::getVariantName)
+                        .collect(Collectors.toList()));
+    }
+
```
Comment (sosa-vaadin) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:655]: This code is related to another feature not intended for this version of flow. Please remove.

**Review by sosa-vaadin — CHANGES_REQUESTED** [review]
There's code from https://github.com/vaadin/flow-components/pull/589 added in this CP. Please remove it since:

- It has nothing to do with the fix addressed in this PR
- It is incomplete and breaks compilation and tests
- It is a feature not intended for this version of flow

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java (hunk 1 above)
Comment (pleku) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java]: Fixed

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java (hunk 2 above)
Comment (pleku) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java]: Fixed

**Review by sosa-vaadin — APPROVED** [review]

**Pull request merged**